crates/uv-auth/src/credentials.rs

```diff
@@ -299,9 +317,20 @@ impl Credentials {
     }
 }
 
+pub fn obfuscate_url_credentials(mut url: Url) -> Url {
```

Comment:
I feel like we may want to introduce a wrapper type for this, like `ObfuscatedUrl` with a different display? It feels weird both to:

- Mutate the given URL
- Return a type that does not encode that there's obfuscation

It feels like too much of a footgun.

Reply:
(The approach implemented here is quite different from the one you did in the middleware, perhaps we should just separate it entirely?)

Reply:
Agreed, I think this should somehow be enforced in the type system (or some kind of `Sensitive` trait).

Reply:
That was just a quick fix in the meantime.

Reply:
But the real solution will use the type system.

Reply:
I've reverted these changes and just left the main `Credentials` fix.